agent/engine/docker_task_engine.go

```diff
@@ -572,10 +583,16 @@ func (engine *DockerTaskEngine) concurrentPull(task *api.Task, container *api.Co
 	defer seelog.Debugf("Released ImagePullDeleteLock after pulling image - %s. Task: %v", container.Image, task)
 	defer ImagePullDeleteLock.RUnlock()
 
-	pullStart := time.Now()
+	// Record the task pull_started_at timestamp
+	pullStart := engine.time().Now()
 	defer func(startTime time.Time) {
 		seelog.Infof("Finished pulling container %v in %s. Task: %v", container.Image, time.Since(startTime).String(), task)
 	}(pullStart)
+	ok := task.SetPullStartedAt(pullStart)
```

Comment:
Since we are setting `PullStartedAt` in both `concurrentPull` as well as `serialPull`methods, can we just set once in  `pullAndUpdateContainerReference`, which is called by `concurrentPull` and `serialPull` ?

Reply:
Ideally we don't want to have duplicate code here, but since we want to set the timestamp after acquired the lock, we can't set in the upper level in `pullAndUpdateContainerReference`.